jenkins/test/testsuite/Jenkinsfile

```diff
@@ -68,20 +70,46 @@ pipeline {
             post() {
                 always {
                     script {
-                        publish()
+                        publishResults()
                     }
                     cleanWs disableDeferredWipeout: true, deleteDirs: true
                 }
             }
         }
     }
+    post() {
+        success {
+            node('Jenkins-Agent-al2-x64-m5xlarge') {
+                script {
+                    def stashed = lib.jenkins.Messages.new(this).get(['Integ-Test'])
+                    publishNotification(":white_check_mark:", "Successful Build", "\n${stashed}")
+                }
+            }
+        }
+        failure {
+            node('Jenkins-Agent-al2-x64-m5xlarge') {
+                script  {
+                    def stashed = lib.jenkins.Messages.new(this).get(['Integ-Test'])
+                    publishNotification(":warning:", "Failed Build", "\n${stashed}")
+                }
+            }
+        }
+    }
 }
 
 /** Publishes test results to S3 **/
-void publish() {
+void publishResults() {
     def artifactPath = "bundle-build/${opensearch_version}/${build_id}/${platform}/${architecture}"
     withAWS(role: 'opensearch-test', roleAccount: "${AWS_ACCOUNT_PUBLIC}", duration: 900, roleSessionName: 'jenkins-session') {
         s3Upload(file: 'test-results', bucket: "${ARTIFACT_BUCKET_NAME}", path: "${artifactPath}/test-results")
     }
-    echo "Uploaded Test Results to ${artifactPath}/test-results"
+    echo "Uploaded Test Results to s3://${ARTIFACT_BUCKET_NAME}/${artifactPath}/test-results"
```

Comment:
Results wont be downloaded via S3. Its not a public bucket. We should add public url to download the results. 
Also can we instruct the component team how they can download the results?

For example (just an example please check actual paths and names):
```
Local cluster logs for your test run: $public_url/${artifactPath}/test-results/<component>/$config/$local-cluster-logs

Your integ-test run logs: $public_url/${artifactPath}/test-results/<component>/$config/$test-outcome

Your overall status: $public_url/${artifactPath}/test-results/<component>/$config/test-outcome.yml
```

No need to specify the component name but just the format so that component teams can download their own logs

Reply:
`component` are not available in JenkinsFile. Is there any way you can think of to extract those?

Reply:
Do not specific the component. Just mention the format in the message so the respective compoenent owner will replace the `component` with say `sql` and can download the logs as they hit above url

Reply:
Wait I dont think even that is doable. We cannot list things in s3. So they can only download the overall status. For all others its a folder with many files. Not sure how to solve that part.  like local-test-cluster logs contains 5-6 files and not a tar.gz or zip that the component owner can download

Reply:
All the buckets are exposed via ci.opensearch.org, so publish those links.

Reply:
Note; Just because s3 listing isn't allow doesn't mean you can't provide a pretty index.html or manifest.yml with urls to all the good stuff.